### ISSUE TITLE
Build failure: SPIRV-Tools SPV_OPERAND_TYPE_FRAGMENT_SHADING_RATE undeclared

### ISSUE THREAD

**Issue by jyamaha** [OpenEmu/OpenEmu#4338]
Summary of issue

# Steps to reproduce

1. Clone GitHub latest version v2.2.1 using command line 
   git clone --recursive https://github.com/OpenEmu/OpenEmu.git
   (note: suggested GUI method also fails replacing this step with compile guide GUI clone w/GiHub Desktop)
2. git submodule update --init --recursive for command-line pull (and also just for grins even for GUI pull)
3. double-click on OpenEmu/OpenEmu.xcworkspace (GUI) - and also ran "/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -verbose -workspace ./OpenEmu.xcworkspace -scheme OpenEmu -configuration Release" for command-line method.  Command-line build fails here.
4. Continued GUI approach: Set scheme to OpenEmu>My Mac
5. Set Product>Build For to "Profiling"  GUI build fails here

# Expected Behavior

1. Expectation is for project to build

# Observed Behavior

1. See error message: ~/Documents/GitHub/OpenEmu/OpenEmu-Shaders/3rdparty/SPIRV-Tools/build/operand.kinds-unified1.inc:1130:4: use of undeclared identifier 'SPV_OPERAND_TYPE_FRAGMENT_SHADING_RATE'; did you mean 'SPV_OPERAND_TYPE_FP_ROUNDING_MODE'?

2. It's true, I grep for "SPV_OPERAND_TYPE_FRAGMENT_SHADING_RATE" and do not see it anywhere in the OpenEmu/OpenEmu-Shaders/3rdparty/SPIRV-Tools/include/spirv-tools/ tree except for the "operand.kinds-unified1.inc" file which seems the only reference to this.

# Debugging Information

HW: MacBook Pro
OS: macOS Catalina 10.15.7
Toolchain: Xcode 12.1
OpenEmu v2.2.1

**Comment by clobber** [comment]
```
$ cd OpenEmu/OpenEmu-Shaders/3rdparty
$ make clean
```

**Comment by jason-rmt** [comment]
make clean of OpenEmu/OpenEmu-Shaders/3rdparty did not fix issue for me.

**Comment by jyamaha** [comment]
make clean of OpenEmu/OpenEmu-Shaders/3rdparty did not resolve issue for me.

**Comment by jyamaha** [comment]
Pushing through, I found that replacing the OpenEmu packaged version of SPIRV-Tools into "OpenEmu/OpenEmu-Shaders/3rdparty/SPIRV-Tools" fixed the compile errors associated with this issue report.

Resolution: 
1) Downloaded .zip file of latest master from https://github.com/KhronosGroup/SPIRV-Tools
2) copy contents in-place to OpenEmu/OpenEmu-Shaders/3rdparty/SPIRV-Tools (I did not delete/clear/touch anything already in the SPIRV-Tools directory in preparation - the cp overwrote any existing files and left alone everything else)
3) Recompiled - did not receive the reported error anymore.

I did receive a different error for two instances of these lines:

```
if (@available(macOS 11.0, *)) {
 //       self.window.toolbarStyle = NSWindowToolbarStyleUnified;
  //      self.window.titlebarSeparatorStyle = NSTitlebarSeparatorStyleLine;
```

And above was my temporary hack to comment out the lines that broke the build.  I haven't researched if this is reported already - or what I should/could do to properly resolve.  It's not a part of this issue other than after fixing the original issue - commenting out these lines (in two .m files) resulted in a successful build.

If someone else runs into a similar problem involving SPIRV-Tools, I suggest pulling from the repository itself as a "patch".

**Comment by clobber** [comment]
I'm on Xcode 12.1; it builds. You either did not `make clean` correctly and also needed to do a Clean Build or your checkout of the `OpenEmu-Shaders` repo was not in sync with master.

We are aware of the issue regarding API availability. That requires Xcode 12.2 with the Big Sur SDK, coming soon.
